### PR TITLE
Inline includes and mark them as jsNode where necessary

### PR DESCRIPTION
**Pull request by b-studios**
This ought to address #611

### CODE REVIEW

File: libraries/common/io/console.effekt [hunk 1]
```diff
@@ -24,26 +24,26 @@ def console[R] { program: () => R / Console }: R = {
 
 namespace js {
 
-  extern js """
+  extern jsNode """
     const readline = require('node:readline');
   """
```
Comment (b-studios) [inline libraries/common/io/console.effekt:29]: Specific to node js and doesn't work in the webbrowser, for now simply disable

File: libraries/common/io/error.effekt [hunk 2]
```diff
@@ -366,8 +366,106 @@ def fromNumber(number: Int): IOError = number match {
 
 namespace internal {
 
+  extern llvm """
+    declare %Int @c_error_number(%Int)
+  """
+
+  extern jsNode """
+    const os = require('os');
```
Comment (b-studios) [inline libraries/common/io/error.effekt:374]: Same, 'os' is only available in node

File: libraries/common/io/error.effekt [hunk 3]
```diff
@@ -366,8 +366,106 @@ def fromNumber(number: Int): IOError = number match {
...
+        [os.constants.errno.EUNATCH]: 219,
+        [os.constants.errno.ESHUTDOWN]: 220
+      };
+      return errnoMap[-errno] || -1; // Default to -1 for unknown error names
+    }
+  """
+
   extern pure def errorNumber(errno: Int): Int =
-    js "errorNumber(${errno})"
+    jsNode "errorNumber(${errno})"
```
Comment (b-studios) [inline libraries/common/io/error.effekt:468]: Using `os`

File: libraries/common/io/filesystem.effekt [hunk 4]
```diff
@@ -139,32 +139,109 @@ def close(fd: File): Unit / Exception[IOError] = {
...
+  extern llvm """
+    declare void @c_fs_open(%Pos, %Pos, %Stack)
+    declare void @c_fs_read(%Int, %Pos, %Int, %Stack)
+    declare void @c_fs_write(%Int, %Pos, %Int, %Stack)
+    declare void @c_fs_close(%Int, %Stack)
   """
 
   extern async def open(path: String, mode: Mode): Int =
-    js "$effekt.callcc(callback => open(${path}, ${mode}, callback))"
+    jsNode "$effekt.callcc(callback => open(${path}, ${mode}, callback))"
```
Comment (b-studios) [inline libraries/common/io/filesystem.effekt:223]: Using `fs` which is only available in node, but not web

File: libraries/common/io/network.effekt [hunk 5]
```diff
@@ -4,7 +4,7 @@ import bytes
 import io
 
 namespace js {
-  extern js """
+  extern jsNode """
     const net = require('node:net');
```
Comment (b-studios) [inline libraries/common/io/network.effekt:8]: `node:net` is only available on node

File: libraries/common/io/error.effekt (hunk 2 above)
Comment (jiribenes) [reply libraries/common/io/error.effekt:374]: We could also use this:
```suggestion
    const os = require('node:os');
```
for consistency.

File: libraries/common/io.effekt [hunk 6]
```diff
@@ -3,15 +3,15 @@ module io
...
-
-
 // Event Loop
 // ----------
 
 type Task[T] = () => T at {io, async, global}
 
+extern llvm """
+  declare void @c_timer_start(%Int, %Stack)
+"""
```
Comment (b-studios) [inline libraries/common/io.effekt:13]: This wouldn't be duplicated, if instead we define a function
`c_spawn(%Stack)` that just calls c_timer_start with `0`.

**Comment by b-studios** [comment]
@phischu I tried to establish the principle that each file only includes its own forward declarations. To this end, and to deduplicate the forward declaration of `c_timer_start`, I introduced `c_yield` which is just `c_timer_start(0, _)`. 

However, while this is a tiny function that just forwards and seems useless, semantically I think it makes sense to distinguish the two.

**Comment by b-studios** [comment]
We can merge this PR in my opinion, the moment CI passes.

**Pull request merged**